homeassistant/components/ssdp/__init__.py

```diff
@@ -251,135 +307,111 @@ async def async_start(self) -> None:
         ]
 
     @core_callback
-    def _async_matching_domains(self, info_with_req: CaseInsensitiveDict) -> set[str]:
+    def _async_matching_domains(self, info_with_desc: CaseInsensitiveDict) -> set[str]:
+        assert self._integration_matchers is not None
         domains = set()
         for domain, matchers in self._integration_matchers.items():
             for matcher in matchers:
-                if all(info_with_req.get(k) == v for (k, v) in matcher.items()):
+                if all(info_with_desc.get(k) == v for (k, v) in matcher.items()):
                     domains.add(domain)
         return domains
 
-    def _async_seen(self, header_st: str | None, header_location: str | None) -> bool:
-        """Check if we have seen a specific st and optional location."""
-        if header_st is None:
-            return True
-        return (header_st, header_location) in self.seen
-
-    def _async_see(self, header_st: str | None, header_location: str | None) -> None:
-        """Mark a specific st and optional location as seen."""
-        if header_st is not None:
-            self.seen.add((header_st, header_location))
-
-    def _async_unsee(self, header_st: str | None, header_location: str | None) -> None:
-        """If we see a device in a new location, unsee the original location."""
-        if header_st is not None:
-            self.seen.discard((header_st, header_location))
-
-    async def _async_process_entry(self, headers: Mapping[str, str]) -> None:
-        """Process SSDP entries."""
-        _LOGGER.debug("_async_process_entry: %s", headers)
-        h_st = headers.get("st")
-        h_location = headers.get("location")
-
-        if h_st and (udn := _udn_from_usn(headers.get("usn"))):
-            cache_key = (udn, h_st)
-            if old_headers := self.cache.get(cache_key):
-                old_h_location = old_headers.get("location")
-                if h_location != old_h_location:
-                    self._async_unsee(old_headers.get("st"), old_h_location)
-            self.cache[cache_key] = headers
-
-        callbacks = self._async_get_matching_callbacks(headers)
-        if self._async_seen(h_st, h_location) and not callbacks:
-            return
-
-        assert self.description_manager is not None
-        info_req = await self.description_manager.fetch_description(h_location) or {}
-        info_with_req = CaseInsensitiveDict(**headers, **info_req)
-        discovery_info = discovery_info_from_headers_and_request(info_with_req)
-
-        _async_process_callbacks(callbacks, discovery_info)
-
-        if self._async_seen(h_st, h_location):
-            return
-        self._async_see(h_st, h_location)
-
-        for domain in self._async_matching_domains(info_with_req):
-            _LOGGER.debug("Discovered %s at %s", domain, h_location)
+    async def _ssdp_listener_callback(
+        self, ssdp_device: SsdpDevice, type: DeviceOrServiceType, source: SsdpSource
```

Comment:
```suggestion
        self, ssdp_device: SsdpDevice, type_: DeviceOrServiceType, source: SsdpSource
```

or something more descriptive